tests/helics/core/HandleManagerTests.cpp

```diff
@@ -0,0 +1,608 @@
+    EXPECT_THROW(h1.addAlias("e2", "endpoint"), std::runtime_error);
+}
+
+TEST(handleManager, duplicateAliasFilter)
+{
+    auto h1 = generateExampleHandleManager();
+    h1.addAlias("f1", "filter");
+    // trying to define 2 interfaces to a single alias
+    EXPECT_THROW(h1.addAlias("f2", "filter"), std::runtime_error);
+}
+
+TEST(handleManager, AliasToInterfacePub)
+{
+    auto h1 = generateExampleHandleManager();
+    // trying to define an alias to a existing target
+    EXPECT_THROW(h1.addAlias("p2", "p1"), std::runtime_error);
+}
+
+TEST(handleManager, AliasToInterfaceTranslator)
+{
+    auto h1 = generateExampleHandleManager();
+    // trying to define an alias to a existing target
+    EXPECT_THROW(h1.addAlias("t2", "t1"), std::runtime_error);
+}
+
+TEST(handleManager, AliasToInterfaceInput)
+{
+    auto h1 = generateExampleHandleManager();
+    // trying to define an alias to a existing target
+    EXPECT_THROW(h1.addAlias("in2", "in1"), std::runtime_error);
+}
+
+TEST(handleManager, AliasToInterfaceEndpoint)
+{
+    auto h1 = generateExampleHandleManager();
+    // trying to define an alias to a existing target
+    EXPECT_THROW(h1.addAlias("e2", "e1"), std::runtime_error);
+}
+
+TEST(handleManager, AliasToInterfaceFilter)
+{
+    auto h1 = generateExampleHandleManager();
+    // trying to define an alias to a existing target
+    EXPECT_THROW(h1.addAlias("f2", "f1"), std::runtime_error);
+}
+
+TEST(handleManager, duplicateAliasDelayed)
+{
+    auto h1 = generateExampleHandleManager();
+    h1.addAlias("p1", "pub");
+    h1.addAlias("p4", "pub");
+    // trying to define 2 interfaces to a single alias delayed
+    EXPECT_THROW(addTestHandle(h1, "p4", InterfaceType::PUBLICATION), std::runtime_error);
+}
+
+TEST(handleManager, duplicateAliasDifferentInterfaceType)
+{
+    auto h1 = generateExampleHandleManager();
+    h1.addAlias("p1", "pub");
+    h1.addAlias("p4", "pub");
+    // it is allowed to have different aliases for different interface types
+    EXPECT_NO_THROW(addTestHandle(h1, "p4", InterfaceType::ENDPOINT));
+}
+
+TEST(handleManager, aliasReciprocity)
+{
+    // aliases should be reciprocal
+    auto h1 = generateExampleHandleManager();
+    h1.addAlias("matthew", "matt");
+    // defining an interface with the name of an alias is allowed as long as the original doesn't
+    // exist, this will create an alias to original interfaceName.
+    EXPECT_NO_THROW(addTestHandle(h1, "matt", InterfaceType::ENDPOINT));
+
+    const auto* ep = h1.getEndpoint("matthew");
+    ASSERT_NE(ep, nullptr);
+    EXPECT_EQ(ep->key, "matt");
+}
+
+TEST(handleManager, aliasCascadeExisting)
+{
+    auto h1 = generateExampleHandleManager();
+    h1.addAlias("p1", "pub");
+    h1.addAlias("pub", "publisher");
+
+    const auto* p1 = h1.getPublication("publisher");
+    ASSERT_NE(p1, nullptr);
+    EXPECT_EQ(p1->key, "p1");
+
+    const auto* p2 = h1.getPublication("pub");
+    ASSERT_NE(p2, nullptr);
+    EXPECT_EQ(p1->key, "p1");
```

Comment:
```suggestion
    EXPECT_EQ(p2->key, "p1");
```

Unless getting a publication alters p1, this is a duplicate of the check on line 524.